File: dcp-solutions/2024-10-28/solution.ex
```elixir
def max_subarray_sum(nums) do
  # Initialize current sum and maximum sum to first element of the array
  current_sum = Enum.sum(enumerable: nums, start: 0)
  max_sum = Enum.sum(enumerable: [num], start: 0)

  # Iterate over the rest of the array
  for num in Enum.drop(nums, 1) do
    # Update current sum to be the current number plus or minus the old current sum
    new_current_sum = max(num, current_sum + num)
    
    # Update maximum sum if current sum is greater
    [max_sum, current_sum] = [new_current_sum, new_current_sum] do
      if new_current_sum > current_sum end
      else
        num
      end
  end

  # Return maximum sum
  max_sum
end
```

File: dcp-solutions/2024-10-28/solution.rb
```ruby
def max_subarray_sum(nums)
  # Initialize variables to track maximum sum and current sum
  max_sum = current_sum = nums.first
  
  # Iterate over array starting from second element
  (1...nums.size).each do |i|
    # Update current sum by adding the current number
    current_sum = [current_sum + nums[i], nums[i]].max
    
    # Update maximum sum if current sum is greater
    max_sum = [max_sum, current_sum].max
  end
  
  max_sum
end
```